pkg/kubectl/cmd/delete.go

```diff
@@ -122,7 +126,7 @@ func NewCmdDelete(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra
 		Long:    delete_long,
 		Example: delete_example,
 		Run: func(cmd *cobra.Command, args []string) {
-			o := deleteFlags.ToOptions(streams)
+			o := deleteFlags.ToOptions(nil, streams)
```

Comment:
Can you just pass `f.DynamicClient()` there and remove the code setting `DeleteOptions.DynamicClient = f.DynamicClient()` from `DeleteOptions.Complete()` method? It took me a while to understand why do you pass `nil` there.
It would at least make the code more unified across multiple use cases, and we can require the `dynamicClient` argument to never be nil...

Reply:
Another option is to pass `f.DynamicClient()` there, as well as keep the code overwriting it in `Complete()` (which will set it to the same value). While redundant, it will at least be consistent.

Reply:
Completing command's options in `Complete` is a pattern we're enforcing throughout entire kubectl code base.

Reply:
@soltysh yes, but we don't invoke `Complete` when we do `DeleteFlags.ToOptions` in `apply` and `replace`, and that's why David had to change the signature of `DeleteFlags.ToOptions` to make sure that dynamic client is initialized for those commands.